### PR TITLE
Update Dockerfile by fixing conda dep

### PR DESCRIPTION
**Pull request by lenjoy**
1) the `bzip2` package should be included for conda, otherwise it relies on the package from the previous docker image;
2) updated the `RUN`s order and added comments for cleaner process and better readability

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -2,40 +2,43 @@ ARG cuda_version=9.0
 ARG cudnn_version=7
 FROM nvidia/cuda:${cuda_version}-cudnn${cudnn_version}-devel
 
+# Install system packages
+RUN apt-get update && \
+    apt-get install -y wget git libhdf5-dev g++ graphviz openmpi-bin libgl1-mesa-glx bzip2
+
```
Comment (HerveNivon) [inline docker/Dockerfile]: Listing the install packages one per line: “sort multine arguments” here (https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#sort-multi-line-arguments)

Comment (lenjoy) [reply docker/Dockerfile]: done

**Comment by lenjoy** [comment]
Thanks @HerveNivon !  I've updated Dockerfile, now the image size has been changed from 8.12GB (before this PR) to 8.05GB (after this PR)

**Comment by HerveNivon** [comment]
@lenjoy hope to save a little bit more. For the next one!

**Comment by fchollet** [comment]
Can this be merged?

**Comment by lenjoy** [comment]
thank you for the review, @HerveNivon , anything else do i need to update?  the primary goal is to fix the dependency and make the code more readable. It's nice to have with file size optimization since the major space is taken in the base docker file.

**Pull request merged**